pkg/planner/cardinality/row_count_column.go

```diff
@@ -177,7 +179,15 @@ func equalRowCountOnColumn(sctx planctx.PlanContext, c *statistics.Column, val t
 		if modifyCount == 0 {
 			return 0, nil
 		}
-		return 1, nil
+		// Use sqrt to create an NDV if one doesn't already exist, or reset to the original NDV
+		if c.Histogram.NDV <= 0 {
```

Comment:
Please add some comments here to explain when this can happen "c.Hist.NDV <= 0".

Reply:
I don't believe that NDV can be < 0 - but this check is here just to be safe. NDV == 0 seems valid if you add a column to a table, with default NULL - Distinct Count in stats_histograms will be 0. If you add the column with default 0 (rather than NULL) - NDV == 1. This is with auto-analyze disabled.